src/session/session_dhandle.c

```diff
@@ -554,6 +554,22 @@ __wt_session_get_btree_ckpt(WT_SESSION_IMPL *session, const char *uri, const cha
      */
 
     is_hs = strcmp(uri, WT_HS_URI) == 0;
+
+    /*
+     * We have already pinned the history store checkpoint dhandle when we open the checkpoint
+     * cursor. No need to resolve and open it again. We only need to lock it here. It is also wrong
+     * to read the metadata with the checkpoint transaction.
+     */
+    if (is_hs && session->hs_checkpoint != NULL) {
```

Comment:
Is it the right place for this fast path? The checks right after on the checkpoint name should still be done.

Reply:
```
if (is_reserved_name && (!is_hs || session->hs_checkpoint == NULL))
```

The following only applies it is not history store.